llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -25093,6 +25097,219 @@ SDValue X86TargetLowering::LowerCTSELECT(SDValue Op, SelectionDAG &DAG) const {
     return DAG.getNode(ISD::TRUNCATE, DL, Op.getValueType(), CtSelect);
   }
 
+  // Handle floating point on i386 without SSE/CMOV (constant-time requirement)
```

Comment:
Can we move some of the operation related to bitcast and read/write from slack slot in `X86TargetLowering::EmitInstrWithCustomInserter` https://github.com/trail-of-forks/llvm-project/blob/ct-llvmorg-20.1.4/llvm/lib/Target/X86/X86ISelLowering.h#L1175

My thoughts are registers we are dealing with in Custom emitter is still virtual. We only deal with operand types and registers of type `f32`, `f64` and `f80` and let the register allocator deal with how they allocate them. If we are dealing with soft-float and need to bitcast we can implement is here.